Assets/MixedRealityToolkit/Utilities/GameObjectManagement/GameObjectCreator.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.GameObjectManagement
```

Comment:
Wouldn't this make more sense as a service?

Reply:
Maybe. This was a direct port from Longbow. I can revisit this to see how it would look as a service and whether or not that would make sense.

Reply:
After looking into the service architecture a little more deeply for MRTK, I don't think that the Game Object Pool is ideal as a service.
The GameObjectPool is meant to be instanced. While you can have as many creators in a GameObjectPool as you want, it doesn't make since to have a GameObjectPool as a single entity to persist across the experience. The logic for this is that different scenes may have their own GameObjectPools that are meant to go away when the scene goes away, as the objects that are being recycled may not make sense to hang around in other scenes.
Additionally you create your own creators in code. While there is a generic one readily available, it will not satisfy object creation in all cases.

Reply:
Note that GameObjectPools are not necessarily scene bound either. They can go away whenever you want them to at any point.

Reply:
It would probably make more sense to write a pooling service from scratch than to rip apart / re-assemble this one.

Reply:
Again, it doesn't necessarily need to be a service. It certainly _could_ be, but the benefit of leaving it as is provides the ability to drop pools as needed. The GameObjectPool is not a singleton or meant to be a single pool. It certainly is _like_ a service, but I'm not convinced it should be unless you can articulate why it is better as a service.

Reply:
Agreed, I'm saying it doesn't seem like the benefits of this tool would survive adaptation to a service.

Reply:
Makes sense to me :)